### PR TITLE
pkg/fileutil: make purgeTest more robust

### PR DESCRIPTION
**Pull request by xiang90**
1. Add a few comments to the test.

2. Provide a more robust way to check the purging
result. Tolerate slow io operations.

Fix https://github.com/coreos/etcd/issues/3747

### CODE REVIEW

File: pkg/fileutil/purge_test.go [hunk 1]
```diff
@@ -47,14 +51,27 @@ func TestPurgeFile(t *testing.T) {
...
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(fnames) > 3 {
+			time.Sleep(10 * time.Millisecond)
+			continue
+		}
+
+		wnames := []string{"7.test", "8.test", "9.test"}
+		if !reflect.DeepEqual(fnames, wnames) {
```
Comment (jonboulle) [inline pkg/fileutil/purge_test.go]: You still need to check this if the loop is broken don't you?

Comment (jonboulle) [inline pkg/fileutil/purge_test.go]: s/broken/completed/

Comment (xiang90) [inline pkg/fileutil/purge_test.go]: my mistake. the break is lost....

File: pkg/fileutil/purge_test.go [hunk 2]
```diff
@@ -47,14 +51,28 @@ func TestPurgeFile(t *testing.T) {
...
+			t.Errorf("filenames = %v, want %v", fnames, wnames)
+		}
+		break
```
Comment (xiang90) [inline pkg/fileutil/purge_test.go]: @jonboulle fixed.

File: pkg/fileutil/purge_test.go (hunk 1 above)
Comment (jonboulle) [inline pkg/fileutil/purge_test.go]: This still isn't what I meant. Why don't you move the equality check OUTSIDE the loop?

Comment (xiang90) [inline pkg/fileutil/purge_test.go]: You are right... Done.

**Comment by xiang90** [comment]
@jonboulle PTAL.

File: pkg/fileutil/purge_test.go [hunk 3]
```diff
@@ -47,14 +51,28 @@ func TestPurgeFile(t *testing.T) {
...
+	// purge routine should purge 7 out of 10 files and only keep the
+	// 3 most recent ones.
+	// wait for purging for at most 100ms.
+	var fnames []string
+	for i := 0; i < 10; i++ {
+		fnames, err = ReadDir(dir)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(fnames) > 3 {
```
Comment (jonboulle) [inline pkg/fileutil/purge_test.go]: I would invert this (`if len(fnames) <= 3 {break}`) to obviate the `continue`, but whatever

Comment (jonboulle) [inline pkg/fileutil/purge_test.go]: I would invert this (`if len(fnames) <= 3 {break}`) to obviate the `continue`, but whatever

**Comment by jonboulle** [comment]
LGTM

File: pkg/fileutil/purge_test.go (hunk 3 above)
Comment (xiang90) [inline pkg/fileutil/purge_test.go]: saving a line is huge. done.

**Pull request merged**